SoftLayer/CLI/object_storage/list_endpoints.py

```diff
@@ -2,26 +2,83 @@
 # :license: MIT, see LICENSE for more details.
 
 import click
-
 import SoftLayer
 from SoftLayer.CLI import environment
 from SoftLayer.CLI import formatting
+from SoftLayer import utils
 
 
 @click.command(cls=SoftLayer.CLI.command.SLCommand, )
+@click.argument('identifier')
 @environment.pass_env
-def cli(env):
+def cli(env, identifier):
     """List object storage endpoints."""
 
     mgr = SoftLayer.ObjectStorageManager(env.client)
-    endpoints = mgr.list_endpoints()
+    endpoints = mgr.list_endpoints(identifier)
+
+    final_end_points = []
 
-    table = formatting.Table(['datacenter', 'public', 'private'])
+    table = formatting.Table(['Legacy', 'EndPoint Type', 'Public/Private', 'Location/Region', 'Url'])
     for endpoint in endpoints:
-        table.add_row([
-            endpoint['datacenter']['name'],
-            endpoint['public'],
-            endpoint['private'],
-        ])
+        data = [endpoint['legacy'], end_point_return(endpoint['region']), public_private(endpoint['type']),
+                location_region(endpoint), endpoint['url']]
```

Comment:
Setting up data as a list of datapoints makes this code hard to follow. My suggestion would be to have data be a dictionary, and `final_end_points` be a list of those dictionaries. 

`add_array_to_table` would take that list of dictionaries, and return a list of lists that can be added to the table.


```python

data = {
    "legacy": endpoint['legacy'],
    "type": end_point_return(endpoint['region']),
    "publicPrivate": public_private(endpoint['type']),
    "region": location_region(endpoint)
    "url": endpoint['url']
}
final_end_points.append(data)
```

then in `add_array_to_table` and `sort_endpoint` you can do something like `array['legacy']` instead of `array[0]`

That way the code should be a bit easier to follow.
Thanks